win32/tests/file_win32_ut/file_win32_ut.c

```diff
@@ -0,0 +1,952 @@
+    uint32_t size = 10;
+    uint64_t position = 5;
+    HANDLE h_event = (HANDLE)44;
+    void* io;
+    PTP_IO captured_ptp_io;
+
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG))
+        .CaptureReturn(&io);
+    STRICT_EXPECTED_CALL(mock_CreateEvent(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG))
+        .SetReturn(h_event);
+    STRICT_EXPECTED_CALL(mock_StartThreadpoolIo(IGNORED_ARG))
+        .CaptureArgumentValue_pio(&captured_ptp_io);
+    STRICT_EXPECTED_CALL(mock_ReadFile(IGNORED_ARG, destination, size, NULL, IGNORED_ARG))
+        .SetReturn(FALSE)
+        .CallCannotFail();
+    STRICT_EXPECTED_CALL(mock_GetLastError())
+        .SetReturn(ERROR_IO_PENDING)
+        .CallCannotFail();
+    STRICT_EXPECTED_CALL(mock_CancelThreadpoolIo(IGNORED_ARG))
+        .ValidateArgumentValue_pio(&captured_ptp_io);
+    STRICT_EXPECTED_CALL(mock_CloseHandle(h_event))
+        .CallCannotFail();
+    STRICT_EXPECTED_CALL(free(IGNORED_ARG))
+        .ValidateArgumentValue_ptr(&io);
+
+    umock_c_negative_tests_snapshot();
+    for (int i = 0; i < umock_c_negative_tests_call_count(); ++i)
+    {
+        if (umock_c_negative_tests_can_call_fail(i))
+        {
+            umock_c_negative_tests_reset();
+            umock_c_negative_tests_fail_call(i);
+
+            ///act
+            FILE_READ_ASYNC_RESULT result = file_read_async(file_handle, destination, size, position, user_callback, NULL);
+
+            ///assert
+            ASSERT_ARE_EQUAL(FILE_READ_ASYNC_RESULT, FILE_READ_ASYNC_ERROR, result);
+        }
+    }
+
+    ///cleanup
+    file_destroy(file_handle);
+}
+
+/*Tests_SRS_FILE_WIN32_43_050: [ file_extend shall return 0. ]*/
+TEST_FUNCTION(file_extend_returns_zero)
+{
+    ///arrange
+
+    ///act
+    int return_val = file_extend(NULL, 0);
+
+    ///assert
+    ASSERT_ARE_EQUAL(int, 0, return_val);
+}
+
+/*Tests_SRS_FILE_WIN32_43_034: [ on_file_io_complete_win32 shall recover the file handle, the number of bytes requested by the user, user_callback and user_context from the context containing overlapped. ]*/
+/*Tests_SRS_FILE_WIN32_43_063: [ on_file_io_complete_win32 shall call GetOverlappedResult to determine if the asynchronous operation was successful. ]*/
+/*Tests_SRS_FILE_WIN32_43_067: [ on_file_io_complete_win32 shall compare number_of_bytes_transferred to the number of bytes requested by the user. ]*/
+/*Tests_SRS_FILE_WIN32_43_066: [ on_file_io_complete_win32 shall call user_callback with is_successful as true if and only if GetOverlappedResult returns true and number_of_bytes_transferred is equal to the number of bytes requested by the user. ]*/
+TEST_FUNCTION(on_file_io_complete_win32_calls_callback_successfully)
+{
+    ///arrange
+    FILE_CB user_callback = mock_user_callback;
+    void* user_context = (void*)20;
+    uint32_t size = 10;
+    PTP_WIN32_IO_CALLBACK captured_callback;
+    LPOVERLAPPED captured_ov;
+
+    get_callback(user_callback, user_context, size, &captured_callback, &captured_ov);

```

Comment:
>get_callback [](start = 4, length = 12)

All the tests for the io complete callback should be done for both read and write. I think you just have it for write so far.